File: leetsync/780-max-chunks-to-make-sorted/max-chunks-to-make-sorted.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int n = arr.length;
        int chunksCount = 0, maxTillNow = 0;
        for (int i = 0; i < n; i++) {
            maxTillNow = Math.max(arr[i], maxTillNow);
            if (maxTillNow == i) {
                chunksCount++;
            }
        }
        return chunksCount;
    }
}

File: leetsync/780-max-chunks-to-make-sorted/max-chunks-to-make-sorted.go
func maxChunksToSorted(arr []int) int {
	n := len(arr)
	chunksCount, maxTillNow := 0, 0
	for i := 0; i < n; i++ {
		if maxTillNow < arr[i] {
			maxTillNow = arr[i]
		}
		if maxTillNow == i {
			chunksCount++
		}
	}
	return chunksCount
}